### PR TITLE
feat(surveys): gate premium features

### PR DESCRIPTION
**Pull request by liyiy**
## Problem

Gate premium surveys features

## Changes
I opted to leave "button" text configurable even though it's under customizations since it's not really just for styling

Maybe too many pay gate blocks?

Screen Shot 2023-10-19 at 10 55 25 AM

Screen Shot 2023-10-19 at 10 05 04 PM
Screen Shot 2023-10-19 at 10 21 45 PM

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

File: frontend/src/scenes/surveys/SurveyAppearance.tsx [hunk 1]
```diff
@@ -134,71 +134,83 @@ export function SurveyAppearance({
 }
 
 export function Customization({ appearance, surveyQuestionItem, onAppearanceChange }: CustomizationProps): JSX.Element {
-    const { whitelabelAvailable } = useValues(surveysLogic)
-    const { featureFlags } = useValues(featureFlagLogic)
+    const { whitelabelAvailable, surveysStylingAvailable } = useValues(surveysLogic)
+
     return (
         <div className="flex flex-col">
+            <PayGateMini feature={AvailableFeature.SURVEYS_STYLING}>
```
Comment (liyiy) [inline frontend/src/scenes/surveys/SurveyAppearance.tsx:141]: I'll add the feature flags once everything else looks good!

**Review by raquelmsmith — CHANGES_REQUESTED** [review]
Can you change where it says "Upgrade to the Scale plan" to just say "Subscribe"? And the button on the paygate should just be "Subscribe", not explore license options 😬 

Seeing it here I'm not a huge fan of all the fields being shown but disabled.... but i think it's a good start and we can iterate on it later. I'm wondering how something more "designed" would come across. Problem for another day!

File: frontend/src/scenes/surveys/SurveyEdit.tsx [hunk 2]
```diff
@@ -944,31 +957,42 @@ export function HTMLEditor({
...
+                                            automaticLayout: true,
+                                            fixedOverflowWidgets: true,
+                                            lineNumbers: 'off',
+                                            glyphMargin: false,
+                                            folding: false,
+                                        }}
+                                    />
+                                ) : (
+                                    <PayGateMini feature={AvailableFeature.SURVEYS_TEXT_HTML}>
+                                        <></>
```
[the remark this answers is not in the document]
Comment (liyiy) [reply frontend/src/scenes/surveys/SurveyEdit.tsx:993]: nope, the component requires children :o 

I can make the children optional, but figured it wasn't a big deal to leave in an empty fragment

File: frontend/src/lib/components/PayGateMini/PayGateMini.tsx [hunk 3]
```diff
@@ -87,7 +98,7 @@ export function PayGateMini({
 
     const featureSummary = FEATURE_SUMMARIES[feature]
     const planRequired = FEATURE_MINIMUM_PLAN[feature]
-    let gateVariant: 'add-card' | 'contact-sales' | 'check-licensing' | null = null
+    let gateVariant: 'add-card' | 'contact-sales' | 'subscribe' | null = null
```
Comment (liyiy) [inline frontend/src/lib/components/PayGateMini/PayGateMini.tsx:101]: this licensing stuff seems pretty outdated so I just switched it out with "subscribe" instead :o

File: frontend/src/scenes/surveys/SurveyAppearance.tsx (hunk 1 above)
Comment (liyiy) [reply frontend/src/scenes/surveys/SurveyAppearance.tsx:141]: pay gate flag added. Basically everything should work as it does currently, and when the flag is enabled, all of this gets gated

**Review by raquelmsmith — APPROVED** [review]
looks good! (minus extra snapshot changes..) don't forget to add the feature flags!

**Comment by liyiy** [comment]
> looks good! (minus extra snapshot changes..) don't forget to add the feature flags!

flags are in! 🤗

**Comment by liyiy** [comment]
All the snapshot changes are due to the changes in `<PayGateMini` with the "licensing" wording being changed to "subscribe" instead

**Pull request merged**